### PR TITLE
Refactor SourceGenerators to use proper equality

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix / refactorings

## What is the new behavior?

Misc adjustments for the reference implementation and wasm split.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.Skia.cs [hunk 1]
```diff
@@ -38,9 +38,25 @@ private protected virtual void OnUnloaded()
...
+		[GeneratedDependencyProperty(DefaultValue = null, ChangedCallback = true)]
+		public static DependencyProperty TransitionsProperty { get; } = CreateTransitionsProperty();
+
+		public TransitionCollection Transitions
+		{
+			get => GetTransitionsValue();
+			set => SetTransitionsValue(value);
+		}
+
+		private void OnTransitionsChanged(DependencyPropertyChangedEventArgs args)
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Interface.Skia.cs:53]: Why generate a callback if private and empty?

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.netstdref.cs [hunk 2]
```diff
@@ -48,7 +48,22 @@ private protected virtual void OnUnloaded()
...
+		[GeneratedDependencyProperty(DefaultValue = null, ChangedCallback = true)]
+		public static DependencyProperty TransitionsProperty { get; } = CreateTransitionsProperty();
+
+		public TransitionCollection Transitions
+		{
+			get => GetTransitionsValue();
+			set => SetTransitionsValue(value);
+		}
+
+		private void OnTransitionsChanged(DependencyPropertyChangedEventArgs args)
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Interface.netstdref.cs:62]: Same here, empty callback

File: src/Uno.UI/UI/Xaml/UIElement.Skia.cs [hunk 3]
```diff
@@ -123,7 +123,7 @@ private void OnChildAdded(UIElement child)
 		internal virtual void OnElementLoaded()
 		{
 			IsLoaded = true;
-			foreach (var innerChild in _children)
+			foreach (var innerChild in _children.ToArray())
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/UIElement.Skia.cs:126]: Not sure which is more performant, `ToArray` or `ToList`? I would suppose `ToList` as it remove the need to trim the array.

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.netstdref.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Interface.netstdref.cs:62]: This one is not used (reference implementation)

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.Skia.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Interface.Skia.cs:53]: Will be improved in next PR.

File: src/Uno.UI/UI/Xaml/UIElement.Skia.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/UIElement.Skia.cs:126]: It's the same underneath. (List creates an array of the right size)

**Pull request merged**